streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -242,16 +242,19 @@ boolean maybeResumeSuspendedTask(final TaskId taskId, final Set<TopicPartition>
                 suspended.remove(taskId);
                 try {
                     task.resume();
+                    transitionToRunning(task, new HashSet<TopicPartition>());
                 } catch (final TaskMigratedException e) {
```

Comment:
As we don't start a new transaction in `task.resume()` anymore, I think a `TaskMigratedException` cannot occur here any longer?